tests/sles4sap/hana_test.pm

```diff
@@ -84,12 +84,14 @@ sub run {
     if (get_var('NVDIMM') and !get_var('SKIP_HANADB_QUERY')) {
         $output = script_output "$hdbsql \"SELECT * FROM M_INIFILE_CONTENTS where file_name = 'global.ini' and section = 'persistence' and key = 'basepath_persistent_memory_volumes'\"";
         my $pmempath = get_var('HANA_PMEM_BASEPATH', "/hana/pmem/$sid");
-        my $nvddevs = get_var('NVDIMM_NAMESPACES_TOTAL', 2);
-        foreach my $i (0 .. ($nvddevs - 1)) {
-            die "hdbsql: HANA not configured with NVDIMM\n\n$output" unless ($output =~ /pmem$i/);
-            assert_script_run "grep -q -w pmem$i /hana/shared/$sid/global/hdb/custom/config/global.ini";
-            assert_script_run "ls $pmempath/pmem$i";
-            assert_script_run "test -n \"\$(ls $pmempath/pmem$i)\"";
+        # Read all configured pmem devices on the system
+        my @pmem_devices_all = split("\n", script_output("find /dev/pmem*"));
+        foreach my $pmem_device (@pmem_devices_all) {
+            $pmem_device =~ s:/dev/(pmem.+):$1:;
```

Comment:
Same question as above.

Reply:
Actually, both of it is intentional. regex match all pmem devices even if they will be moved another structure in /dev/

So, it will work with all of folowing:
/dev/pmemx
/dev/pmemxy
/dev/pmem/xy

Reply:
But you are right. It too wide. We definitely want it closed to one word.